### PR TITLE
Adapt virtcontainers to fit different runtime expectations

### PR DESCRIPTION
**Pull request by sboeuf**
Some adjustments related to what can expect different runtimes, and particularly OCI runtime, needs to be made to get everything working as expected.

### CODE REVIEW

File: pod.go [hunk 1]
```diff
@@ -235,6 +235,13 @@ type PodStatus struct {
 	ContainersStatus []ContainerStatus
 }
 
+// ShimInfo gathers information needed by a shim, returned from a proxy.
+type ShimInfo struct {
+	Token string
+	IP    string
+	Port  string
+}
```
Comment (sameo) [inline pod.go]: One Pod could have many shims associated to it. `IP` and `Port` make sense, but I don't think `Token` does.

Comment (sboeuf) [inline pod.go]: Yes you're right. But in case I use a token list, how can I know (from runtime implementation) how to differentiate the tokens ?

Comment (sameo) [inline pod.go]: You don't need to store the shim tokens at the PodConfig level. It's going to be up to the proxy to store those and also their mappings with e.g. the hyperstart seqnums.

**Comment by sameo** [comment]
LGTM

**Comment by sameo** [comment]
@sboeuf Could you please rebase and push ?

**Pull request merged**